geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/NullRedisSet.java

```diff
@@ -58,13 +59,13 @@ public int scard() {
   }
 
   @Override
-  public long sadd(List<byte[]> membersToAdd, Region<RedisKey, RedisData> region, RedisKey key) {
-    region.create(key, new RedisSet(membersToAdd));
+  public long sadd(List<byte[]> membersToAdd, RegionProvider regionProvider, RedisKey key) {
+    regionProvider.getLocalDataRegion().create(key, new RedisSet(membersToAdd));
```

Comment:
getLocalDataRegion is more expensive than getDataRegion and in this case we are using it to do a create. The region returned by getLocalDataRegion only allows local reads but write ops are unconstrained. Since create is a write op I think you can just change this to getDataRegion